tests/test_indexes.py

```diff
@@ -61,6 +61,10 @@ def test_index_astype() -> None:
         pd.DataFrame,
     )
 
+    df = pd.DataFrame({"a": [1, 2, 3]})
+    mi = pd.MultiIndex.from_product([["x", "y"], df.columns])
+    mi = pd.MultiIndex.from_product([["x", "y"], pd.Series([1, 2])])
```

Comment:
can you add the `check(assert_type(` pattern here?

Reply:
Fixed